go/libraries/doltcore/mvdata/data_loc.go

```diff
@@ -89,7 +89,7 @@ type DataLocation interface {
 	Exists(ctx context.Context, root *doltdb.RootValue, fs filesys.ReadableFS) (bool, error)
 
 	// NewReader creates a TableReadCloser for the DataLocation
-	NewReader(ctx context.Context, root *doltdb.RootValue, fs filesys.ReadableFS, opts interface{}) (rdCl table.SqlRowReader, sorted bool, err error)
+	NewReader(ctx context.Context, dEnv *env.DoltEnv, opts interface{}) (rdCl table.SqlRowReader, sorted bool, err error)
```

Comment:
yeah i agree, but the alternative was adding databases and config to this interface. Passing 4 fields from env felt dumb, If you want a one-off table reader it might make sense to just have the dEnv